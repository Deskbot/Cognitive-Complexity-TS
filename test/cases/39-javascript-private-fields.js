class A {
    #b() {
        return this.#b();
    }

    get #c() {
        return true && false;
    }

    static #d() {
        return true && false;
    }

    static #e = () => { return true && false; }

    static #f = function () { return true && false; }
}