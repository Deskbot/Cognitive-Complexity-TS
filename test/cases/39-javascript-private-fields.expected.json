{
    "39-javascript-private-fields.js": {
        "score": 5,
        "inner": [
            {
                "name": "A",
                "score": 5,
                "line": 1,
                "column": 1,
                "inner": [
                    {
                        "name": "#b",
                        "score": 1,
                        "line": 2,
                        "column": 5
                    },
                    {
                        "name": "#c",
                        "score": 1,
                        "line": 6,
                        "column": 5
                    },
                    {
                        "name": "#d",
                        "score": 1,
                        "line": 10,
                        "column": 5
                    },
                    {
                        "name": "#e",
                        "score": 1,
                        "line": 14,
                        "column": 17
                    },
                    {
                        "name": "#f",
                        "score": 1,
                        "line": 16,
                        "column": 17
                    }
                ]
            }
        ]
    }
}